### PR TITLE
Node: Add RouterDump and DirectTransportDump types

### PR DESCRIPTION
**Pull request by jmillan**
NOTE: I'll be checking which more API types are missing and create the corresponding PRs.

### CODE REVIEW

File: node/src/Router.ts [hunk 1]
```diff
@@ -128,6 +128,41 @@ export type PipeToRouterResult =
 	pipeDataProducer?: DataProducer;
 };
 
+export type RouterDump = {
```
Comment (ibc) [inline node/src/Router.ts]: { in next line please.

File: node/src/utils.ts [hunk 2]
```diff
@@ -91,6 +91,25 @@ export function parseVector<Type>(
...
+/**
+ * Parse flatbuffers vector of StringString into the corresponding array.
+ */
+export function parseStringStringVector(
+	binary: any, methodName: string
+): { key: string; value: string }[]
+{
+	const array: { key: string; value: string }[] = [];
+
+	for (let i=0; i<binary[`${methodName}Length`](); ++i)
```
Comment (ibc) [inline node/src/utils.ts]: i = 0 with spaces as everywhere else. Not sure why ESLint doesn't complain.

File: node/src/utils.ts [hunk 3]
```diff
@@ -147,3 +166,28 @@ export function parseUint32StringVector(
...
+/**
+ * Parse flatbuffers vector of StringStringArray into the corresponding array.
+ */
+export function parseStringStringArrayVector(
+	binary: any, methodName: string
+): { key: string; values: string[] }[]
+{
+	const array: { key: string; values: string[] }[] = [];
+
+	for (let i=0; i<binary[`${methodName}Length`](); ++i)
```
Comment (ibc) [inline node/src/utils.ts]: Same

File: node/src/utils.ts [hunk 4]
```diff
@@ -147,3 +166,28 @@ export function parseUint32StringVector(
...
+	{
+		const kv = binary[methodName](i)!;
+		const values: string[] = [];
+
+		for (let i2=0; i2<kv.valuesLength(); ++ i2)
```
Comment (ibc) [inline node/src/utils.ts]: Same

**Review by ibc — APPROVED** [review]
Cosmetic things requested. BTW hell, I was about to do this today but ok.

File: node/src/utils.ts (hunk 2 above)
Comment (jmillan) [reply node/src/utils.ts]: Eslint rule set.

**Comment by jmillan** [comment]
Converting to draft as I'm going to handle the rest of the cases here.

File: node/.eslintrc.js [hunk 5]
```diff
@@ -183,6 +183,7 @@ const eslintConfig =
 			}
 		],
 		'space-in-parens' : [ 2, 'never' ],
+		'space-infix-ops' : [ 2, { 'int32Hint': false } ],
```
Comment (ibc) [inline node/.eslintrc.js:186]: Good. I'm adding it in other projects.

**Review by ibc — APPROVED** [review]
Why a PR draft yet? IMHO this is ready.

**Comment by ibc** [comment]
> Why a PR draft yet? IMHO this is ready.

Ah no, a test is failing... but this cannot be related to this PR...

https://github.com/versatica/mediasoup/actions/runs/6654790668/job/18083561359?pr=1195

**Comment by ibc** [comment]
> Converting to draft as I'm going to handle the rest of the cases here.

Which rest of cases?

**Comment by ibc** [comment]
> > Why a PR draft yet? IMHO this is ready.
> 
> Ah no, a test is failing... but this cannot be related to this PR...
> 
> https://github.com/versatica/mediasoup/actions/runs/6654790668/job/18083561359?pr=1195

Ok, indeed this PR is modifying the failing test.

**Comment by nazar-pc** [comment]
@ibc too many notifications, GitHub is not an instant messenger, try to combine multiple messages into one and multiple comments into reviews if possible :pray:

File: node/src/tests/test-Consumer.ts [hunk 6]
```diff
@@ -1009,27 +1009,27 @@ test('consumer.close() succeeds', async () =>
...
 		]));
 
-	dump = await transport2.dump();
+	const transportDump = await transport2.dump();
 
-	dump.consumerIds = dump.consumerIds.sort();
+	transportDump.consumerIds = transportDump.consumerIds.sort();
 
-	expect(dump)
+	expect(routerDump)
```
Comment (ibc) [inline node/src/tests/test-Consumer.ts]:
```suggestion
	expect(transportDump)
```

Comment (ibc) [reply node/src/tests/test-Consumer.ts]: commiting.

**Comment by jmillan** [comment]
Bonus track from `flatbuffers`. All public methods are typed now.

**Review by ibc — APPROVED** [review]

**Pull request merged**